4_Zadania_dodatkowe/6.InfiniteSlider/js/app.js

```diff
@@ -0,0 +1,65 @@
+$(document).ready(function () {
+
+    var $prevButton = $('#prevPicture');
+    var $nextButton = $('#nextPicture');
+
+    var $liList = $('li');
+
+    var $ul = $('ul');
+
+    var visibleImageIndex = 0;
+    var imageWidth = $('img').width()+4;   // I tried with outerwidth, css(width) etc but still I have to add 4 because of the space between images; reset of all elements also didn't help
+
+    var $firstLi = $liList.first().clone();
+    var $lastLi = $liList.last().clone();
+    $ul.append($firstLi);
+    $ul.prepend($lastLi);
+
+    $ul.css('right', imageWidth+'px').css('width', imageWidth*8+'px');
+
+    function move(parameter) {
+        if (parameter===1) {
+            visibleImageIndex++;
+        }
+        $ul.animate({
+                'right': imageWidth * visibleImageIndex,
+            },
+            {
+                complete: function () {
+                    if (parameter===1) {
+                        if (visibleImageIndex > $liList.length) {
+                            visibleImageIndex = 1;
+                            $ul.css('right', imageWidth);
+                        }
+                    } else if (parameter===-1) {
```

Comment:
see line 29 and line 34
these `parameter` conditions are needed I think
it's not possible that `visibleImageIndex > $liList.length` but `parameter` is `-1`

I think you what you should do here is to:
1. calculate new value of `visibleImageIndex` if it corresponds to cloned elements (and you got this logic right! lines 30-31 and lines 35-36)
2. if `visibleImageIndex` was reassigned then do `$ul.css('right', imageWidth * visibleImageIndex);`